src/FrameReflower/Page.php

```diff
@@ -174,31 +174,26 @@ function reflow(BlockFrameDecorator $block = null)
      * Check for callbacks that need to be performed when a given event
      * gets triggered on a page
      *
-     * @param string $event the type of event
-     * @param Frame $frame  the frame that event is triggered on
+     * @param string $event The type of event
+     * @param Frame  $frame The frame that event is triggered on
      */
-    protected function _check_callbacks($event, $frame)
+    protected function _check_callbacks(string $event, Frame $frame): void
     {
         if (!isset($this->_callbacks)) {
             $dompdf = $this->_frame->get_dompdf();
             $this->_callbacks = $dompdf->getCallbacks();
             $this->_canvas = $dompdf->getCanvas();
         }
 
-        if (is_array($this->_callbacks) && isset($this->_callbacks[$event])) {
+        if (isset($this->_callbacks[$event])) {
+            $fs = $this->_callbacks[$event];
             $info = [
                 0 => $this->_canvas, "canvas" => $this->_canvas,
                 1 => $frame,         "frame"  => $frame,
             ];
-            $fs = $this->_callbacks[$event];
+
             foreach ($fs as $f) {
-                if (is_callable($f)) {
-                    if (is_array($f)) {
-                        $f[0]->{$f[1]}($info);
-                    } else {
-                        $f($info);
-                    }
-                }
+                $f($info);
```

Comment:
Just an FYI. I believe the method was _supposed_ to support both a function and an array with an object/string combination representing a callable class method. I don't know if that was ever documented anywhere. And it looks like we pretty much never really supported it since one of the conditions for adding a callback was that it was callable 😕 

I think this change is fine since we never supported the other method and, really, who would use it anyway.

Reply:
Hm, but `is_callable` and the `$callable()` syntax both support an array with object/string! They also support an array with class name/string to call static methods, which would previously fail because of the `is_array` check. See https://www.php.net/manual/en/language.types.callable.php for reference.

Support for the object/string combination was documented in the `setCallbacks` doc block (_… set to a function or object plus method to be called_).

Reply:
ah ha ha shows what _I_ know, I obviously haven't been using this feature of PHP very much